include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -43,7 +43,9 @@ namespace __internal
 // walk1
 //------------------------------------------------------------------------
 
-template <typename _BackendTag, typename _ExecutionPolicy, typename _ForwardIterator, typename _Function>
+template <__par_backend_hetero::access_mode __acc_mode = __par_backend_hetero::access_mode::read_write,
```

Comment:
Thinking about the template defaults, they make sense for the common case across `__pattern_walk` 1,2,3. But is having different default template arguments for `__pattern_walk1` from the others confusing?

Reply:
I think it does makes sense to have these defaults (read_write) for `__pattern_walk1`, partially because this has been the only mode prior to this PR for `__pattern_walk1`.  However, I wouldn't object to simply requiring the access mode always (no default).

We could have the default be `write` & `no_init`, like the others, but I would worry a little about that because it changes the semantics of walk1 if no arguments are given.

While looking at this, I realize that `uninitialized_fill` and `uninitialized_value_construct`, `uninitialized_default_construct` also dont need `read_write`, but currently use it because they go through a wrapper of pattern_walk1.  I should probably propagate these template options up to all the wrappers.